go-controller/pkg/ovn/ovn.go

```diff
@@ -211,17 +216,26 @@ func (oc *DefaultNetworkController) ensureRemoteZonePod(oldPod, pod *kapi.Pod, a
 			return fmt.Errorf("addPodExternalGW failed for remote pod %s/%s: %v", pod.Namespace, pod.Name, err)
 		}
 	}
+	if kubevirt.IsPodLiveMigratable(pod) {
+		return kubevirt.EnsureRemoteZonePodAddressesToNodeRoute(oc.controllerName, oc.watchFactory, oc.nbClient, oc.lsManager, pod, ovntypes.DefaultNetworkName)
+	}
 	return nil
 }
 
 // removePod tried to tear down a pod. It returns nil on success and error on failure;
 // failure indicates the pod tear down should be retried later.
 func (oc *DefaultNetworkController) removePod(pod *kapi.Pod, portInfo *lpInfo) error {
 	if oc.isPodScheduledinLocalZone(pod) {
-		return oc.removeLocalZonePod(pod, portInfo)
+		if err := oc.removeLocalZonePod(pod, portInfo); err != nil {
+			return err
+		}
+	} else {
+		if err := oc.removeRemoteZonePod(pod); err != nil {
+			return err
+		}
 	}
 
-	return oc.removeRemoteZonePod(pod)
+	return kubevirt.CleanUpLiveMigratablePod(oc.nbClient, oc.watchFactory, pod)
```

Comment:
Would it add clarity if we enclosed this in a `if kubevirt.IsPodLiveMigratable(pod)`

Reply:
We are already doing the following inside the function, and the name make clear what is trying to remove.
```
// This pod is not part of ip migration so we don't need to clean up        
    if !IsPodLiveMigratable(pod) {                                              
        return nil                                                              
    }    
```